content/departments/cloud/technical-docs/v2.0/restore_process.md

```diff
@@ -0,0 +1,106 @@
+# Restore a managed instance v2.0 when it is completely gone
+
+<span class="badge badge-note">SOC2/CI-110</span>
+
+## Restoring Cloud SQL
+
+Use cases:
+
+- Cloud SQL data is corrupted by a broken database migration
+- Cloud SQL data is deleted
+
+### Restore from automated backup
+
+> Below process is derived from [GCP documentation](https://cloud.google.com/sql/docs/postgres/backup-recovery/restoring#gcloud)
+
+The restortion process will be performed with `gcloud`. Learn more about [why not terraform?](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#restore_backup_context).
+
+Locate the SQL instance, note the name of the instance as `SQL_INSTANCE`
+
+```sh
+gcloud sql instances list --project $PROJECT_ID
+```
+
+List all backups, note the name of the latest (or the one right before database state is corrupted) **SUCCESSFUL** backup as `SQL_BACKUP_ID`
+
+```sh
+gcloud sql backups list --instance $SQL_INSTANCE --project $PROJECT_ID
+```
+
+Restore the backup to the current instance.
+
+> NOTE: **IMPORTANT** This will override all current data with the backup.
+
+```sh
+gcloud sql backups restore $SQL_BACKUP_ID --restore-instance $SQL_INSTANCE --project $PROJECT_ID
+```
+
+## Restore GKE cluster application(s)
+
+Use cases (tested scenarios):
+
+- GKE cluster was deleted
+- application namespaces was deleted
+- single application was deleted
+- PV from single application was deleted
+
+### How GKE backup/restore works
+
+Backup and restore uses [native GKE mechanism](https://cloud.google.com/kubernetes-engine/docs/add-on/backup-for-gke/concepts/backup-for-gke).
+
+> NOTE: **IMPORTANT** Backup has to be enabled during [creation process](./creation_process.md#enable-backup)
+
+1. List backups
+
+```sh
+mi2 instance backup list --slug <CUSTOMER> -e [dev|prod]
+```
+
+2. Create on-demand backup
```

Comment:
Why are we creating a backup during restore - is this supposed to be used when restore goes wrong?

Reply:
No, I just used it during my tests to have fresh backup. I remove it from here, b/c we have daily backups same as for CloudSQL, so on-demand can go to FAQ.